examples/autoqasm/6_Customize_gate_calibrations.ipynb

```diff
@@ -0,0 +1,404 @@
+    "    pulse.play(q0_rf_frame, wf)\n",
+    "    pulse.shift_frequency(q0_rf_frame, 321047.14178613486)\n",
+    "    pulse.barrier(\"$0\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9300f0f8",
+   "metadata": {},
+   "source": [
+    "After the gate calibration is defined, it is ready to be attach to a quantum program. Say, we want to attach `my_defcal` to the main quantum program `my_program`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 53,
+   "id": "bae7731a",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "OPENQASM 3.0;\n",
+      "rx(pi / 2) $0;\n",
+      "rz(0.123) $1;\n",
+      "bit __bit_0__;\n",
+      "__bit_0__ = measure $0;\n"
+     ]
+    }
+   ],
+   "source": [
+    "@aq.main\n",
+    "def my_program():\n",
+    "    rx(\"$0\", np.pi/2)\n",
+    "    rz(\"$1\", 0.123)\n",
+    "    measure(\"$0\")\n",
+    "\n",
+    "main_program = my_program()\n",
+    "print(main_program.to_ir())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c7579516",
+   "metadata": {},
+   "source": [
+    "We can use the `with_calibrations` method of the program which creates a new program with `my_defcal` attached, leaving the original intact. You can experiment with different gate calibrations on the same program without needing to redefine the main program every time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 54,
+   "id": "6b7cc158",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "OPENQASM 3.0;\n",
+      "extern drag_gaussian(duration, duration, float[64], float[64], bool) -> waveform;\n",
+      "waveform wf_drag_gaussian_1 = drag_gaussian(24.0ns, 2.547965400864ns, 2.370235498840002e-10, 0.293350447987059, false);\n",
+      "defcal rx(pi / 2) $0 {\n",
+      "    barrier $0;\n",
+      "    shift_frequency(q0_rf_frame, -321047.14178613486);\n",
+      "    play(q0_rf_frame, wf_drag_gaussian_1);\n",
+      "    shift_frequency(q0_rf_frame, 321047.14178613486);\n",
+      "    barrier $0;\n",
+      "}\n",
+      "rx(pi / 2) $0;\n",
+      "rz(0.123) $1;\n",
+      "bit __bit_0__;\n",
+      "__bit_0__ = measure $0;\n"
+     ]
+    }
+   ],
+   "source": [
+    "custom_program = main_program.with_calibrations(my_defcal)\n",
+    "print(custom_program.to_ir())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f038b8c8",
+   "metadata": {},
+   "source": [
+    "You can also define a gate calibration with variable parameters. For variable parameters, instead of setting it in the decorator, you set it as a arguments of the decorated function. The parameters must have type hint. The type hint for qubits is `aq.Qubit`, and the type hint for angles is `float`. Let's define another gate calibration. This time, we define for `rz` gate on qubit `$1` and a variable angle `angle`."
```

Comment:
"For variable parameters, instead of setting it in the decorator, you set it as a arguments of the decorated function." Why? By providing an explanation, the customer will be able to remember what they've read more easily

Reply:
```suggestion
    "You can also define a gate calibration with variable parameters. For variable parameters, instead of setting it in the decorator, you set it as a arguments of the decorated function. The parameters must have a type hint. The type hint for qubits is `aq.Qubit`, and the type hint for angles is `float`. Let's define another gate calibration. This time, we define a calibration for the `rz` gate on qubit `$1` and a variable angle `angle`."
```